hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -0,0 +1,126 @@
+
+import static com.hederahashgraph.api.proto.java.HederaFunctionality.TokenAccountWipe;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.mono.context.properties.GlobalDynamicProperties;
+import com.hedera.node.app.service.mono.txns.TransitionLogicLookup;
+import com.hedera.node.app.service.mono.utils.accessors.TokenWipeAccessor;
+import com.hedera.node.app.workflows.common.InsufficientBalanceException;
+import com.hedera.node.app.workflows.common.PreCheckException;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.SignatureMap;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The {@code IngestChecker} contains checks that are specific to the ingest workflow */
+public class IngestChecker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(IngestChecker.class);
+
+    private final TransitionLogicLookup transitionLogic;
+    private final GlobalDynamicProperties dynamicProperties;
+
+    /**
+     * Constructor of the {@code IngestChecker}
+     *
+     * @param transitionLogic a {@link TransitionLogicLookup}
+     * @param dynamicProperties the {@link GlobalDynamicProperties}
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    public IngestChecker(
+            @NonNull final TransitionLogicLookup transitionLogic,
+            @NonNull final GlobalDynamicProperties dynamicProperties) {
+        this.transitionLogic = requireNonNull(transitionLogic);
+        this.dynamicProperties = requireNonNull(dynamicProperties);
+    }
+
+    /**
+     * Checks a transaction for semantic errors
+     *
+     * @param txBody the {@link TransactionBody}
+     * @param functionality the {@link HederaFunctionality} of the transaction
+     * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws PreCheckException if a semantic error was discovered. The contained {@code
+     *     responseCode} provides the error reason.
+     */
+    public void checkTransactionSemantic(
+            @NonNull final TransactionBody txBody, @NonNull final HederaFunctionality functionality)
+            throws PreCheckException {
+        final ResponseCodeEnum errorCode;
+        if (functionality == TokenAccountWipe) {
+            // TODO: Not really sure why TokenAccountWipe gets special treatment
+            errorCode =
+                    TokenWipeAccessor.validateSyntax(
+                            txBody.getTokenWipe(),
+                            dynamicProperties.areNftsEnabled(),
+                            dynamicProperties.maxBatchSizeWipe());
```

Comment:
Because we tried to move to sub accessors for extending alias support. But we expect them to be sub classes of `TransactionMetadata` . So we won't need to have special case for that

Reply:
Cool, I will replace this as soon as the new functionality is available.